BREAKING CHANGE: Support multiple containers for testing both unit- and integration tests - Fixes #204

johlju opened the pull request:
### About breaking change 
**Note: It is probably only SqlServerDsc is affected by this breaking change.**

I labeled this as a breaking change according to guidelines as it removes a parameter that was previously used for starting containers, and changes the behavior of parameter `RunTestInOrder` as it now handles the container logic. Also the decoration `[Microsoft.DscResourceKit.IntegrationTest(OrderNumber = 1)]` need to be moved from the configuration file to the integration test file.
See more in the change log.

### Changes

- Changes to DscResource.Tests
  - Added new helper function Get-DscTestContainerInformation to read the container
    information in a particular PowerShell script test file (.Tests.ps1).
  - BREAKING CHANGE: For those repositories that are using parameter `RunTestInOrder`
    for the helper function `Invoke-AppveyorTestScriptTask` the
    decoration `[Microsoft.DscResourceKit.IntegrationTest(OrderNumber = 1)]` need
    to move from the configuration file to the test file. This was done since unit
    tests does not have configuration files, and also to align the ability to
    define the order and the container information using the same decoration.
    It is also natural to have the decoration in the test files since those are
    the scripts that are actually run in order.
  - BREAKING CHANGE: The parameter `RunInDocker` is removed in helper function
    `Invoke-AppveyorTestScriptTask`. Using parameter `RunTestInOrder` will now
    handle running tests in a container, but only if at least on test is decorated
    using `[Microsoft.DscResourceKit.IntegrationTest()]` or
    `[Microsoft.DscResourceKit.UnitTest()]`, together with the correct named arguments.
  - Added support for the default shared module to run unit test and integration test
    in a container by decorating each test file with either
    `[Microsoft.DscResourceKit.IntegrationTest()]` or
    `[Microsoft.DscResourceKit.UnitTest()]`.
- Changes to DcsResource.Container
  - Now has support for verifying if image, with or without
    a tag, exist locally or needs to be pulled from Docker Hub.
  - Now shows the correct localized message when downloading
    an image.
  - If 'latest' tag is used on the image name, 'docker pull' will be called to
    make sure the local revision of 'latest' is actually the latest revision on
    Docker Hub. If it isn't, then the latest image will be pulled from Docker Hub.

### Tests
This are updated as the test runs finishes. 

* Repository using shared default model without paramater `RunTestInOrder` (PSDscResource):
* Repository using "harness" model (SharePointDsc): 
* Repository using shared default model with parameter `RunTestInOrder` and without containers (SqlServerDsc): https://ci.appveyor.com/project/johlju/sqlserverdsc/build/9.0.387.0
* Repository using shared default model with 2 containers:
* Repository using shared default model with 2 containers - build worker fail:
* Repository using shared default model with 2 containers - 1st container fail:
* Repository using shared default model with 2 containers - 2nd container fail:
* Repository using shared default model with 2 containers - both containers fail:
* Repository using shared default model with 2 containers - both containers and build worker fail:

johlju commented:
@PlagueHO could you review this one when you got a chance? A lot of texts that needs proof reading too 😄

PlagueHO commented:
@johlju - I'll get onto this tomorrow night after work (getting a bit late tonight). Looks like there is a conflict in here to fix as well.

johlju commented:
@PlagueHO no worries! Just happy you take the time! 😄  Thanks for reminding me of the conflict! I totally forgot I probably needed to do that when Katie merged your PR :)

PlagueHO commented:
Hi @Johlju - some awesome stuff there mate! I'm only part way through the review, but I'll try and finish it tomorrow night after work!

---

Reviewed 5 of 8 files at r1.
Review status: 5 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[Microsoft.DscResourceKit.cs, line 7 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfuYpyA79xJsfM3EQh:-LAfuYpyA79xJsfM3EQi:b-gaxesr) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/Microsoft.DscResourceKit.cs#L7)):*
> ```Smalltalk
> // either unit tests or integration tests for a DSC resource.
> //
> // This is examples on how to write the decoration in a .Tests.ps1 file.
> ```

"This is examples on how to" -> "The following are examples of"

---

*[README.md, line 394 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfyYyh5FPFmYzvE9aF:-LAfyYyh5FPFmYzvE9aG:b8izncs) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L394)):*
> ```
> This make it possible to run integration tests and unit tests
> ```
This makes it possible to run integration tests and unit tests

---

*[README.md, line 395 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfyadrDGYJZV-9GaI6:-LAfyadrDGYJZV-9GaI7:b-sfyiei) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L395)):*
> ```Markdown
> The same parameter `RunTestInOrder` can also be use to run unit tests or integration
> tests in a container. This make it possible to run integration tests and unit tests
> in parallell on the same build worker.
> ```

parallell -> parallel

---

*[README.md, line 398 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfykgf35ILAFB6vWvd:-LAfykgf35ILAFB6vWve:b-a072hj) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L398)):*
> ```Markdown
> The common tests will by default always be run on the AppVeyor build worker.
> 
> To run a test in a container, the test must be decorated with either attribute `Microsoft.DscResourceKit.IntegrationTest` or `Microsoft.DscResourceKit.UnitTest`
> ```

This line seems really long? Can it be shortened?

Also, could read:
To run a test in a container, the test must be decorated with the attribute `Microsoft.DscResourceKit.IntegrationTest` or `Microsoft.DscResourceKit.UnitTest`

---

*[README.md, line 420 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfzKQr7XsfTA2SNvTt:-LAfzKQr7XsfTA2SNvTu:b-hxz8d0) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L420)):*
> ```Markdown
> 
> * **ContainerName**: The name of the container. If the same container name is used
>   in multiple test they will be run in the same container.
> ```

test -> tests

Also, will the tests be run in the same container in series or in parallel? I know they will be in series, but it might be worth making this clear.

---

*[README.md, line 422 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfz_VS5-t0mmT5Bcmq:-LAfz_VS5-t0mmT5Bcmr:b84b17h) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L422)):*
> ```Markdown
>   in multiple test they will be run in the same container.
> * **ContainerImage**: The name of the container image to use for the container.
>   This should use the normal Docker format for specifying a Docker image. If the
> ```

Might be worth adding an example: E.g. microsoft/windowsservercore:latest - just to make it clear. Besides, this is the most likely container image they'll be using.

---

*[README.md, line 430 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg1We_B-mZ3hoBb1RN:-LAg1We_B-mZ3hoBb1RO:b-fn8vlm) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L430)):*
> ```Markdown
>   used.*
> 
> #### Example
> ```

It could be worth showing an example of running an integration test in a container without an order number.

---

*[README.md, line 432 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfzy427TO0LV2KkfTk:-LAfzy427TO0LV2KkfTl:bf3tj5t) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L432)):*
> ```Markdown
> #### Example
> 
> Example showing how the integration test file could look like to run the test in
> ```

Could be changed to read:
This example shows how the integration test file would look if the tests should be run in a container and also run before other integration tests.

---

*[README.md, line 444 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg-Zpb6Ne3wH9vrTED:-LAg-Zpd4hxVTE3kpb1W:bxjvag8) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L444)):*
> ```Markdown
> Example showing how the unit test file could look like to run the test in
> a container. This should be put a the top of the unit test script file.
> *Unit test does not support ordered testing at this time.*
> ```

Unit test does not > Unit tests do not.

English is such an illogical language :cry:

---

*[README.md, line 633 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg-t3pAvLG74q66Zfh:-LAg-t3pAvLG74q66Zfi:bsl5o4v) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L633)):*
> ```Markdown
>   decoration `[Microsoft.DscResourceKit.IntegrationTest(OrderNumber = 1)]` need
>   to move from the configuration file to the test file. This was done since unit
>   tests does not have configuration files, and also to align the ability to
> ```

does not > do not

---

*[README.md, line 648 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg02D0ACLj6AIq3Unx:-LAg02D0ACLj6AIq3Uny:b-yae3mr) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L648)):*
> ```Markdown
> * DcsResource.Container
>   * Now has support for verifying if image, with or without
>     a tag, exist locally or needs to be pulled from Docker Hub.
> ```

exist -> exists

---

*[TestHelper.psm1, line 1408 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg0NTY1yo99KFXHvk0:-LAg0NTY1yo99KFXHvk1:b-yae3mr) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1408)):*
> ```PowerShell
>         if found.
>         If the attribute 'Microsoft.DscResourceKit.IntegrationTest' or
>         'Microsoft.DscResourceKit.UnitTest' exist with at least one of the named
> ```

exist -> exists

---

*[TestHelper.psm1, line 1440 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg0ukh7pmhVdYEpXr2:-LAg0ukh7pmhVdYEpXr3:bvdtxte) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1440)):*
> ```PowerShell
> 
>     <#
>         Will always return $null if the attribute 'Microsoft.DscResourceKit.IntegrationTest'
> ```

I think this comment shouldn't be here.

---

*[TestHelper.psm1, line 1465 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg1kGI48bCWOWNqK0m:-LAg1kGI48bCWOWNqK0n:b-sakrpy) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1465)):*
> ```PowerShell
>         $findAttributeArgumentFilter = {
>             $args[0] -is [System.Management.Automation.Language.NamedAttributeArgumentAst] `
>             #-and $args[0].ArgumentName -eq 'OrderNumber'
> ```

I don't think this line should be here.

---

*[DscResource.Container/DscResource.Container.psm1, line 174 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg2p0t9WjDU39mBshF:-LAg2p0t9WjDU39mBshG:bwjx60d) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/DscResource.Container/DscResource.Container.psm1#L174)):*
> ```PowerShell
>     if ($ImageName -match ':latest$' -or (-not $dockerImages.Contains($ImageName)))
>     {
>         Write-Info ($script:localizedData.DownloadingImage -f $ImageName)
> ```

Should be named parameter

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-:-LAg3I4G-h5bGI2wdvIV:bpqd1jp)*
<!-- Sent from Reviewable.io -->

johlju commented:
Thanks a lot for reviewing this one, and finding all the spelling errors! :smile: I think I got all the comments so far.

---

Review status: 1 of 8 files reviewed at latest revision, 15 unresolved discussions.

---

*[Microsoft.DscResourceKit.cs, line 7 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfuYpyA79xJsfM3EQh:-LAgFyLm6r2hm9YIVTGl:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/Microsoft.DscResourceKit.cs#L7)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

"This is examples on how to" -> "The following are examples of"
</blockquote></details>

Done.

---

*[README.md, line 394 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfyYyh5FPFmYzvE9aF:-LAgG0yk1VsyKd8YxoNf:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L394)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

> ```
> This make it possible to run integration tests and unit tests
> ```
This makes it possible to run integration tests and unit tests
</blockquote></details>

Done.

---

*[README.md, line 395 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfyadrDGYJZV-9GaI6:-LAgG1lW3x0EOqf7URDA:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L395)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

parallell -> parallel
</blockquote></details>

Done.

---

*[README.md, line 398 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfykgf35ILAFB6vWvd:-LAgG3J2BhNqkLhnhvM9:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L398)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

This line seems really long? Can it be shortened?

Also, could read:
To run a test in a container, the test must be decorated with the attribute `Microsoft.DscResourceKit.IntegrationTest` or `Microsoft.DscResourceKit.UnitTest`

</blockquote></details>

Done.

---

*[README.md, line 420 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfzKQr7XsfTA2SNvTt:-LAgG5mo6Y-u-8-blBCa:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L420)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

test -> tests

Also, will the tests be run in the same container in series or in parallel? I know they will be in series, but it might be worth making this clear.
</blockquote></details>

Done.

---

*[README.md, line 422 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfz_VS5-t0mmT5Bcmq:-LAgG7xO-AI4t9k2iqT3:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L422)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Might be worth adding an example: E.g. microsoft/windowsservercore:latest - just to make it clear. Besides, this is the most likely container image they'll be using.
</blockquote></details>

Done.

---

*[README.md, line 430 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg1We_B-mZ3hoBb1RN:-LAgGBq1EVQYG_ZpOxBG:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L430)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

It could be worth showing an example of running an integration test in a container without an order number.
</blockquote></details>

Done.

---

*[README.md, line 432 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAfzy427TO0LV2KkfTk:-LAgGGNg6V6Jk1B4xrJc:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L432)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Could be changed to read:
This example shows how the integration test file would look if the tests should be run in a container and also run before other integration tests.
</blockquote></details>

Done.

---

*[README.md, line 444 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg-Zpb6Ne3wH9vrTED:-LAgGHMk3REB2YVv-18b:bt8i7pk) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L444)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Unit test does not > Unit tests do not.

English is such an illogical language :cry:
</blockquote></details>

Done. I think it's those small variations that is hardest in all languages :smile: 

---

*[README.md, line 633 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg-t3pAvLG74q66Zfh:-LAgGj35E9nT8HUYTV7q:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L633)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

does not > do not
</blockquote></details>

Done.

---

*[README.md, line 648 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg02D0ACLj6AIq3Unx:-LAgGZJJ8Etd3K_WIS8P:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/README.md#L648)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

exist -> exists
</blockquote></details>

Done.

---

*[TestHelper.psm1, line 1408 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg0NTY1yo99KFXHvk0:-LAgGn-OBhhSN45jIW33:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1408)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

exist -> exists
</blockquote></details>

Done.

---

*[TestHelper.psm1, line 1440 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg0ukh7pmhVdYEpXr2:-LAgGnrP1BMv7o-0qf0n:b-j19gst) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1440)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I think this comment shouldn't be here.
</blockquote></details>

Done. Yes it shouldn't be there, it's covered in the comment-based help under `.OUTPUTS`.

---

*[TestHelper.psm1, line 1465 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg1kGI48bCWOWNqK0m:-LAgGxDe2rEGxK7-TZOk:b-55um9d) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/TestHelper.psm1#L1465)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I don't think this line should be here.
</blockquote></details>

Done. Debug stuff that was left in :)

---

*[DscResource.Container/DscResource.Container.psm1, line 174 at r1](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-LAg2p0t9WjDU39mBshF:-LAgH5BkEaa53Kgi8a5c:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/df3f338171d6b8a28c2b4d963a661e8274906cd4/DscResource.Container/DscResource.Container.psm1#L174)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Should be named parameter
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/dscresource.tests/216#-:-LAgH649BfC9QbQMoueq:brqs7r2)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@Johlju - super code. Really just grammatical corrections in code comments and a few mislabelled 'it's'.

---

Reviewed 2 of 8 files at r1, 1 of 1 files at r3, 2 of 4 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[AppVeyor.psm1, line 561 at r4](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAi6mI7096JSKvJQEUB:-LAqnk5x7n9FxRp7zxO1:b-h9lbp1) ([raw file](https://github.com/powershell/dscresource.tests/blob/13c88b9077f5c2b70ccc10d4c9bf4d17e1ec6311/AppVeyor.psm1#L561)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I have a logic problem here. This is suppose to set the remaining script code files to use for code coverage for the main build worker tests. But this is set in the foreach-loop, looping through each container, so this is set to all the remaining script files that isn't tested in the last container. 

I saw this today when I finally merged fixes into SqlServerDsc allowing my to test with all the tests using multiple containers. I saw that I got a lot of missed hits in the build worker for tests that wasn't tested there. :sweat_smile: 

@PlagueHO Maybe don't have time to fix this during the week, if not I will fix this next weekend. Should be an easy fix, so if you could review around this problem I appreciate it. :smile: 
</blockquote></details>

No problem! Good catch though. I'll keep this comment open.

---

*[AppVeyor.psm1, line 138 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkW8xDIee8BnR69V5:-LAqkW8xDIee8BnR69V6:b-thhhxx) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L138)):*
> ```PowerShell
>         attribute, then that test will run among the last tests, after all the
>         integration test with a specific order has run.
>         This will also enables running unit tests and integration tests in a
> ```

This will also enable

---

*[AppVeyor.psm1, line 268 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlJnI21GwGUKLGqfD:-LAqlJnI21GwGUKLGqfE:b-slm23b) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L268)):*
> ```PowerShell
>                     coverage for the tests that are run by the build worker.
>                     If the container testing functionality is used (parameter
>                     'RunTestInOrder'), then the container logic will handle the
> ```

Might be worth stating that the 'RunTestInOrder' parameter is passed and not null.

---

*[AppVeyor.psm1, line 269 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlqSr2PdFXVpOeH3q:-LAqlqSr2PdFXVpOeH3r:b-ehc3iy) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L269)):*
> ```PowerShell
>                     If the container testing functionality is used (parameter
>                     'RunTestInOrder'), then the container logic will handle the
>                     gathering of the files used for a correct CodeCoverage
> ```

a -> the

---

*[AppVeyor.psm1, line 270 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlum_2nh8TtLVWAn-:-LAqlum_2nh8TtLVWAn0:b-lepzgj) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L270)):*
> ```PowerShell
>                     'RunTestInOrder'), then the container logic will handle the
>                     gathering of the files used for a correct CodeCoverage
>                     for the tests run in the build worker.
> ```

for -> from

---

*[AppVeyor.psm1, line 272 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqmJi4-SVuHQru-bX5:-LAqmJi4-SVuHQru-bX6:bnk3h0t) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L272)):*
> ```PowerShell
>                     for the tests run in the build worker.
> 
>                     Although, the array $codeCoveragePaths will be used in the
> ```

I'm not sure this statement is that useful - or possibly I'm misunderstanding it. Is there a way of clarifying it?

---

*[AppVeyor.psm1, line 463 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqn5gvBzqJrG8BtnA0:-LAqn5gvBzqJrG8BtnA1:bnher4t) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L463)):*
> ```PowerShell
>                     Import-Module -Name (Join-Path -Path $PSScriptRoot -ChildPath 'DscResource.Container')
> 
>                     Write-Info -Message 'Using one or more Docker Windows container to run tests.'
> ```

container -> containers

---

*[AppVeyor.psm1, line 504 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqnGkz0tvzCUDX3sBK:-LAqnGkz0tvzCUDX3sBL:buvakgm) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L504)):*
> ```PowerShell
>                                 code coverage in the build worker.
> 
>                                 The $codeCoveragePaths array is build at the begin
> ```

build at the begin -> built at the beginning

---

*[AppVeyor.psm1, line 599 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqntmEESgbxcD35NrP:-LAqntmEESgbxcD35NrQ:b-nlv37p) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L599)):*
> ```PowerShell
>                     }
> 
>                     Write-Info -Message 'One or more containers are started. Build worker will continue run other tests.'
> ```

One or more containers are started. Build worker will continue run other tests - > One or more containers have been started. Build worker will continue to run other tests.

---

*[AppVeyor.psm1, line 626 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoEShECmbgYvTtQSY:-LAqoEShECmbgYvTtQSZ:b-yhqr3w) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L626)):*
> ```PowerShell
>                 <#
>                     Finally add integration tests that can run unordered.
>                     These tests does not have an order number, and does not have
> ```

and does not have -> and do not have

---

*[AppVeyor.psm1, line 764 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoPdIBTndzMmXeWz1:-LAqoPdK98iaXc5cernh:b8yg0e2) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L764)):*
> ```PowerShell
>                         else
>                         {
>                             Write-Warning -Message 'The container did not report any test result! This could indicate an error in the container.'
> ```

Should this just warn or should it actually throw an error. Theoretically we wouldn't want the CI to pass if this occurred wouldn't we?

---

*[AppVeyor.psm1, line 897 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoeQ48hA1BCLpXbKI:-LAqoeQ48hA1BCLpXbKJ:b-tg9g5w) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L897)):*
> ```PowerShell
>                     else
>                     {
>                         # The container was the first to reporting code coverage.
> ```

reporting -> report

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 748 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhGvp5NW_oXZI2KiD:-LAqhGvp5NW_oXZI2KiE:b-d5iln8) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L748)):*
> ```PowerShell
> 
>                     Assert-MockCalled -CommandName 'docker' `
>                     -ParameterFilter $mockDockerImagesWithTag_ParameterFilter -Exactly -Times 0 -Scope It
> ```

Line should be indented.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 760 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhRO8EVA5-1eKKIzJ:-LAqhRO8EVA5-1eKKIzK:b-yae3mr) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L760)):*
> ```PowerShell
>                 }
> 
>                 Context 'When the container image already exist' {
> ```

exist -> exists

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 806 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhlBCAcOBd7qQ88hn:-LAqhlBCAcOBd7qQ88ho:b-d5iln8) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L806)):*
> ```PowerShell
> 
>                     Assert-MockCalled -CommandName 'docker' `
>                     -ParameterFilter $mockDockerImagesWithTag_ParameterFilter -Exactly -Times 1 -Scope It
> ```

Line should be indented.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 818 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhvjQ9PW3otVKqfgf:-LAqhvjQ9PW3otVKqfgg:b-yae3mr) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L818)):*
> ```PowerShell
>                 }
> 
>                 Context 'When the container image already exist' {
> ```

exist -> exists

---

*[Tests/Unit/TestHelper.Tests.ps1, line 57 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjB2nBivzjhWXpoWv:-LAqjB2nBivzjhWXpoWw:bz1voax) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L57)):*
> ```PowerShell
>         }
> 
>         Context 'When configuration file does contain a attribute and with the correct named attribute argument, and there are also another named argument' {
> ```

contain a attribute -> contain an attribute
there are also -> there is also

---

*[Tests/Unit/TestHelper.Tests.ps1, line 63 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjLar7lchq6MWel1O:-LAqjLar7lchq6MWel1P:bmblth7) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L63)):*
> ```PowerShell
>             }
>         }
>     }
> ```

Need line feed after line.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 70 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjPIA7fxKTc_jXggm:-LAqjPIA7fxKTc_jXggn:b-arnqpr) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L70)):*
> ```PowerShell
>         }
> 
>         Context 'When the test is a integration tests' {
> ```

When the test is a integration tests -> When the test is an integration test

---

*[Tests/Unit/TestHelper.Tests.ps1, line 119 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjj9I67Wkq8fRZVdA:-LAqjj9I67Wkq8fRZVdB:b-69mqbs) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L119)):*
> ```PowerShell
> 
>             Context 'When configuration file does contain a attribute and with only the correct named attribute argument ''ContainerName''' {
>                 It 'Should not return any value' {
> ```

Maybe should say: Should return the Mock Container Name and not return a Container Image.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 128 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjzjl8EruFm3_XhZP:-LAqjzjl8EruFm3_XhZQ:bp3jnnp) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L128)):*
> ```PowerShell
> 
>             Context 'When configuration file does contain a attribute and with the correct named attribute arguments' {
>                 It 'Should not return any value' {
> ```

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 137 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkCRQ0YQc4u0HcwU0:-LAqkCRQ0YQc4u0HcwU1:bt5a1x) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L137)):*
> ```PowerShell
> 
>             Context 'When configuration file contain more attributes than just the correct named attribute arguments' {
>                 It 'Should not return any value' {
> ```

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 189 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkLF21nwCcjjhWjeA:-LAqkLF21nwCcjjhWjeB:b-69mqbs) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L189)):*
> ```PowerShell
> 
>             Context 'When configuration file does contain a attribute and with only the correct named attribute argument ''ContainerName''' {
>                 It 'Should not return any value' {
> ```

Maybe should say: Should return the Mock Container Name and not return a Container Image.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 198 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkHLM5yFXRIA0XpWK:-LAqkHLM5yFXRIA0XpWL:bt5a1x) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L198)):*
> ```PowerShell
> 
>             Context 'When configuration file does contain a attribute and with the correct named attribute arguments' {
>                 It 'Should not return any value' {
> ```

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LAqkSmt-lZUOA4kNLYv:bael3qr)*
<!-- Sent from Reviewable.io -->

johlju commented:
Missed a couple of review comments, and will also fix the logic error, will return soon.  :smiley: 

---

Review status: 5 of 8 files reviewed at latest revision, 24 unresolved discussions, some commit checks failed.

---

*[AppVeyor.psm1, line 138 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkW8xDIee8BnR69V5:-LBpOIFqAXpyyWJSLadh:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L138)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

This will also enable
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 268 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlJnI21GwGUKLGqfD:-LBpOM0D1iccLdtfb9kJ:bbgjip0) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L268)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Might be worth stating that the 'RunTestInOrder' parameter is passed and not null.
</blockquote></details>

Done. Rewrote this entire paragraph, hopefully more informative this time. :slightly_smiling_face: 

---

*[AppVeyor.psm1, line 269 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlqSr2PdFXVpOeH3q:-LBpOcbb4K62MC7vMeBe:b-mqn3z8) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L269)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

a -> the
</blockquote></details>

Done. See previous comment

---

*[AppVeyor.psm1, line 270 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqlum_2nh8TtLVWAn-:-LBpOdQD9u6Sh20dx0X0:b-mqn3z8) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L270)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

for -> from
</blockquote></details>

Done. See previous comment

---

*[AppVeyor.psm1, line 272 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqmJi4-SVuHQru-bX5:-LBpOdno8_rHMqHFNo6C:b-mqn3z8) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L272)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I'm not sure this statement is that useful - or possibly I'm misunderstanding it. Is there a way of clarifying it?
</blockquote></details>

Done. See previous comment

---

*[AppVeyor.psm1, line 504 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqnGkz0tvzCUDX3sBK:-LBpOx5i2_bwN6psizQV:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L504)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

build at the begin -> built at the beginning
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 599 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqntmEESgbxcD35NrP:-LBpP1qo18daioR_AxHV:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L599)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

One or more containers are started. Build worker will continue run other tests - > One or more containers have been started. Build worker will continue to run other tests.
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 626 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoEShECmbgYvTtQSY:-LBpP3XH3Q4c5EH7Shle:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L626)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

and does not have -> and do not have
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 764 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoPdIBTndzMmXeWz1:-LBpP4m99gDGbI3FIji9:bh0ujbn) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L764)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Should this just warn or should it actually throw an error. Theoretically we wouldn't want the CI to pass if this occurred wouldn't we?
</blockquote></details>

Done. Agreed. Could been because of debugging during development I didn't want it to throw.  It uploads the logs as artifacts before this so I think if this fails, we could you stop the CI. Changed it to throw here instead, and modified the text.

---

*[AppVeyor.psm1, line 897 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoeQ48hA1BCLpXbKI:-LBpPpeBCRo5-U4n-lRk:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L897)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

reporting -> report
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 748 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhGvp5NW_oXZI2KiD:-LBpPrD32mb8ymjeViW0:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L748)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Line should be indented.
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 760 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhRO8EVA5-1eKKIzJ:-LBpPrq26v6JreH6be3b:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L760)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

exist -> exists
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 806 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhlBCAcOBd7qQ88hn:-LBpPsLd-b9u_m1pWrgW:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L806)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Line should be indented.
</blockquote></details>

Done.

---

*[Tests/Unit/DscResource.Container.Tests.ps1, line 818 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqhvjQ9PW3otVKqfgf:-LBpPsoM5q-v6kOBK3-g:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/DscResource.Container.Tests.ps1#L818)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

exist -> exists
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 63 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjLar7lchq6MWel1O:-LBpQ6TC2TWtj8uKlIP_:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L63)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Need line feed after line.
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 70 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjPIA7fxKTc_jXggm:-LBpQ8iz1qdNcue6o4sk:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L70)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

When the test is a integration tests -> When the test is an integration test
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 119 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjj9I67Wkq8fRZVdA:-LBpQBCu1GV9JO9kdv77:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L119)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Maybe should say: Should return the Mock Container Name and not return a Container Image.
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 128 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjzjl8EruFm3_XhZP:-LBpQBgVBZBWlKzDX7XZ:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L128)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 137 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkCRQ0YQc4u0HcwU0:-LBpQCB78tpDNOx2fdLh:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L137)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 189 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkLF21nwCcjjhWjeA:-LBpQC_L2PoGeaEYvb-E:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L189)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Maybe should say: Should return the Mock Container Name and not return a Container Image.
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 198 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqkHLM5yFXRIA0XpWK:-LBpQD6DDOwE4nJtmFl3:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L198)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Maybe should say: Should return the Mock Container Name and Mock Container Image Name.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LBpQKtw5oLC2ywTEe3v:bbp9jpg)*
<!-- Sent from Reviewable.io -->

johlju commented:
All good now! Ready for review again! :smiley: 

---

Review status: 3 of 8 files reviewed at latest revision, 24 unresolved discussions, some commit checks failed.

---

*[AppVeyor.psm1, line 561 at r4](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAi6mI7096JSKvJQEUB:-LBpjg-x9v20dX9h1ZZP:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/13c88b9077f5c2b70ccc10d4c9bf4d17e1ec6311/AppVeyor.psm1#L561)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

No problem! Good catch though. I'll keep this comment open.
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 463 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqn5gvBzqJrG8BtnA0:-LBpje7f05cfoSOv1Nzh:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L463)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

container -> containers
</blockquote></details>

Done.

---

*[Tests/Unit/TestHelper.Tests.ps1, line 57 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqjB2nBivzjhWXpoWv:-LBpjnJq3fX2WdxD7feu:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/Tests/Unit/TestHelper.Tests.ps1#L57)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

contain a attribute -> contain an attribute
there are also -> there is also
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LBpjpovEnSHmvCuX-qw:bg6kar4)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Almost there! Just some tiny grammatical ones left! :grin:

---

Reviewed 5 of 5 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[AppVeyor.psm1, line 626 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoEShECmbgYvTtQSY:-LBtYjzIAD97e5sWvXOu:b-qzg6iy) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L626)):*
> ```
>  These tests does not have an order number, and do not have
> ```
-> 
These tests do not have an order number, and do not have

---

*[AppVeyor.psm1, line 274 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtXoyA8DUN4SV_MAoB:-LBtXoyA8DUN4SV_MAoC:bl56isa) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L274)):*
> ```PowerShell
>                     the switch parameter 'RunTestInOrder', then the container
>                     logic will handle the gathering of the files that are used
>                     for a correct CodeCoverage.
> ```

for a -> for the 

---

*[AppVeyor.psm1, line 276 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtXy6CBQbjVhVpplzF:-LBtXy6CBQbjVhVpplzG:b-j94dpa) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L276)):*
> ```PowerShell
>                     for a correct CodeCoverage.
>                     Although even if the container logic is handling code coverage,
>                     it will still depend on the below array $codeCoveragePaths to
> ```

depend on the below array $codeCoveragePaths to -> depend on the $codeCoveragePaths array below to

---

*[AppVeyor.psm1, line 278 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtYG-4DDuctG301DbF:-LBtYG-64-IdAKCDjqV6:b-h8827m) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L278)):*
> ```PowerShell
>                     it will still depend on the below array $codeCoveragePaths to
>                     determine the possible paths that can be used to gather the
>                     files that can be used for code coverage.
> ```

that can be -> that will be

---

*[AppVeyor.psm1, line 561 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtYTcz4Y4hy94oKW8P:-LBtYTcz4Y4hy94oKW8Q:b-7dou83) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L561)):*
> ```PowerShell
> 
>                             <#
>                                 The container get the module files it needs for
> ```

get -> gets

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LBtZL828bcMWD34tI2w:b-uod2pn)*
<!-- Sent from Reviewable.io -->

johlju commented:
Hopefully I got them all :)

---

Review status: 7 of 8 files reviewed at latest revision, 5 unresolved discussions.

---

*[AppVeyor.psm1, line 626 at r5](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LAqoEShECmbgYvTtQSY:-LBta-Zs4VmuG0rHMPoL:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/7c039e8b30e2421f5529f37122e5a1047dc0f536/AppVeyor.psm1#L626)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

> ```
>  These tests does not have an order number, and do not have
> ```
-> 
These tests do not have an order number, and do not have
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 274 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtXoyA8DUN4SV_MAoB:-LBt_sSw8JHl0FpSXvAU:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L274)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

for a -> for the 
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 276 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtXy6CBQbjVhVpplzF:-LBt_vS-3VbjXQJJVxjg:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L276)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

depend on the below array $codeCoveragePaths to -> depend on the $codeCoveragePaths array below to
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 278 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtYG-4DDuctG301DbF:-LBt_wQGDw7XhFNDtYWi:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L278)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

that can be -> that will be
</blockquote></details>

Done.

---

*[AppVeyor.psm1, line 561 at r6](https://reviewable.io/reviews/powershell/dscresource.tests/216#-LBtYTcz4Y4hy94oKW8P:-LBt_xqw3xkZ3pe_P6tW:b-896fix) ([raw file](https://github.com/powershell/dscresource.tests/blob/aa1832acd42f2eb0183c4998eb2cc77d3f671675/AppVeyor.psm1#L561)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

get -> gets
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LBta35j3VjSoIvFpKei:b-55vgkw)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Woop! Got them all! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/dscresource.tests/216#-:-LBtck6iC0XoOBdD4RQI:b-klrm87)*
<!-- Sent from Reviewable.io -->

johlju commented:
@PlagueHO Thanks! Finally this is merged. 😃
